src/aap_eda/core/management/commands/purge_log_records.py

```diff
@@ -0,0 +1,139 @@
+            help=(
+                "Specify the activation ids which you want to clear their "
+                "records (e.g., ActivationID1 ActivationID2)"
+            ),
+        )
+        parser.add_argument(
+            "--activation-names",
+            nargs="+",
+            type=str,
+            dest="activation-names",
+            help=(
+                "Specify the activation names which you want to clear their "
+                "records (e.g., ActivationName1 ActivationName2)"
+            ),
+        )
+        parser.add_argument(
+            "--date",
+            dest="date",
+            action="store",
+            required=True,
+            help=(
+                "Purge records older than this date from the database. "
+                "The cutoff date in YYYY-MM-DD format"
+            ),
+        )
+
+    def purge_log_records(
+        self, ids: list[int], names: list[str], cutoff_date: str
+    ) -> None:
+        new_instance_logs = []
+        if not bool(ids) and not bool(names):
+            instances = models.RulebookProcess.objects.all()
+        else:
+            instances = models.RulebookProcess.objects.filter(
+                Q(activation__id__in=ids) | Q(activation__name__in=names),
+            )
+
+        if not instances.exists():
+            logger.info("No records has been found for purging.")
+            return
+
+        for instance in instances:
+            new_instance_logs.append(
+                self.clean_instance_logs(instance, cutoff_date)
+            )
+
+        new_instance_logs = [
+            item for item in new_instance_logs if item is not None
+        ]
+        if len(new_instance_logs) > 0:
+            models.RulebookProcessLog.objects.bulk_create(new_instance_logs)
+
+            logger.info(f"Log records older than {cutoff_date} are purged.")
+
+    def clean_instance_logs(
+        self,
+        instance: models.RulebookProcess,
+        cutoff_date: str,
+    ) -> Union[models.RulebookProcessLog, None]:
+        try:
+            ts = datetime.strptime(cutoff_date, "%Y-%m-%d")
+        except ValueError as exc:
+            raise ValueError(
+                "Timestamp must be in 'YYYY-MM-DD' format."
+            ) from exc
+
+        log_records = models.RulebookProcessLog.objects.filter(
+            activation_instance=instance,
+            log_timestamp__lt=int(ts.timestamp()),
+        )
+        if not log_records.exists():
+            return
+
+        log_records.delete()
+        dt = f"{datetime.now().strftime('%Y-%m-%d %H:%M:%S')}"
+
+        return models.RulebookProcessLog(
+            log=(
+                f"All log records older than {cutoff_date} are purged at {dt}."
+            ),
+            activation_instance_id=instance.id,
+            log_timestamp=int(ts.timestamp()),
+        )
+
+    @transaction.atomic
+    def handle(self, *args, **options):
+        input_ids = [int(id) for id in options.get("activation-ids") or []]
```

Comment:
I think you don't want to have a nested list and this is:
```suggestion
        input_ids = [int(id) for id in options.get("activation-ids")] or []
```
Anyway I think there is no need to call `int(id)` since argparse validates and cast the value for you, so it should be enough with 
```suggestion
        input_ids = options.get("activation-ids") or []
```
as you already do with `input_names`